### DISCUSSION TITLE
Why repaint when the mouse is not hovering my app?

### DISCUSSION THREAD

**Discussion by attran1** [emilk/egui#5284] [Q&A]
Hello all,

I use eframe to create a native app.  I put print statement in the update() function to check the repaint feature.
When the mouse is hover my app, I see a bunch of print statement.  That is expected.
When the mouse is outside of the app boundary, the repaint still happen only when the mouse last click is on the app.  If I click anywhere outside of the app and make my mouse movement, there are no prints.
Is there a way to stop completely the repaint if the mouse is not hovering my app?

Anthony

**Comment by lucasmerlin** [comment]
My guess is that winit is still firing mouse move events when the window is focused and the mouse is moving outside of the window, and every winit event causes a repaint. Not sure what you could do about that 🤔. Maybe egui needs some way to filter events that you're not interested in?

**Comment by attran1** [comment]
Hi,

Should egui be updated to avoid this problem?
If I have to handle on my app, how would I do?  I am new to Rust.

Anthony

**Comment by FroogalTheDragon** [comment]
This sounds like https://github.com/emilk/egui/issues/3931

**Reply by attran1** [reply]
my code does not use request_repaint.  Here is my update function.  
Note that I have to use ui.collapsing to mask all those events.  The Left Panel doesn't have this instruction and I keep receiving those events there.

Anthony

    fn update(&mut self, ctx: &Context, _frame: &mut eframe::Frame) {

        TopBottomPanel::top("menu").show(ctx, |ui: &mut Ui|{
            ui.menu_button("File", |ui|{
                if ui.button("Save").clicked(){println!("saved");} 
                if ui.button("Quit").clicked(){ctx.send_viewport_cmd(egui::ViewportCommand::Close);} 
            });
        });

        SidePanel::left("input")
            .default_width(300.0)
            .width_range(90.0..=300.0)
            .show(ctx, |ui: &mut Ui|{
                println!("left panel");
                ui.heading("left panel");
        });

        SidePanel::right("output")
            .default_width(300.0)
            .width_range(90.0..=300.0) 
            .show(ctx, |ui: &mut Ui|{
                egui::ScrollArea::vertical().show(ui, |ui|{
                    ui.collapsing("right top panel", |ui| {
                        println!("right top panel");
                        ui.heading("right top panel");
                            });
                    ui.collapsing("right bottom panel", |ui| {                        
                        println!("right bottom panel");
                        ui.heading("right bottom panel");
                            });
                });
            });

        CentralPanel::default().show(ctx, |ui| {
            ui.collapsing("center panel", |ui| {
                println!("center panel"); 
            });
        });
    }

**Comment by barries** [comment]
A minimal reproduction of this is to create an `eframe::App` with, then run it and wiggle the mouse outside the app window.

```
impl eframe::App for IotasoftUiApp { 
    fn update(&mut self, _ctx: &egui::Context, _frame: &mut eframe::Frame) {
        println!("!");
    }
}
```
